lib/bike_brigade/messaging.ex

```diff
@@ -26,20 +26,24 @@ defmodule BikeBrigade.Messaging do
     Repo.all(SmsMessage)
   end
 
+  # Private function to get the latest message subquery
+  defp get_latest_message_subquery() do
+    subquery(
+      from m in SmsMessage,
+        where: m.rider_id == parent_as(:rider).id,
+        order_by: [desc: m.sent_at],
+        limit: 1
+    )
+  end
+
   @doc """
   Returns pairs of riders and the last message, sorted by last message sent
   """
   def list_sms_conversations(opts \\ []) do
     query =
       from r in Rider,
         as: :rider,
-        inner_lateral_join:
-          latest_message in subquery(
-            from m in SmsMessage,
-              where: m.rider_id == parent_as(:rider).id,
-              order_by: [desc: m.sent_at],
-              limit: 1
-          ),
+        inner_lateral_join: latest_message in ^get_latest_message_subquery(),
```

Comment:
Small nit but I bumped ecto in #246 and it now gives warnings that the join doesn't specify an `on` condition. 
When you rebase you may need to ad an `on: true` to these joins

Reply:
Done!